### PR TITLE
Create recommended sort for unauthenticated users.

### PR DESCRIPTION
**Pull request by J-coder13**
+ Create Recommended part 2 algorithim
    +  sorts by a weighted some of distance score and reputation score

 + Create Sort by time
    + Displays most recent listings posted for users who are unauthenticated and do not share/give a location

 + Improve DistanceMatrixOBJ
    + Made improvements so that functions can stand alone.

### CODE REVIEW

File: src/main/java/com/google/sps/data/Listing.java [hunk 1]
```diff
@@ -66,6 +71,7 @@ public Listing(String description, String howToHelp, String imageURL,
     this.views = views;
     this.vote = vote;
     this.website = website;
+    this.distance = 0;
```
Comment (zrideaux) [inline src/main/java/com/google/sps/data/Listing.java:74]: I think this variable would make more sense to call `distanceScore` so it stays consistent with `reputationScore` and doesn't get confused with physical distance in meters

File: src/main/java/com/google/sps/sort/recommended/RecommendedSort.java [hunk 2]
```diff
@@ -48,17 +53,63 @@
...
+  public static List<Listing> sortByDistanceAndReputation(List<Listing> listings, String userLocation) 
+      throws IOException {
+
+    //Grab all locations for distance calculation  
+    String[] locations = new String[listings.size()];
+    for (int i = 0; i < listings.size(); i++) {
+      locations[i] = listings.get(i).getLocation(); 
+    }
+    
+    DistanceMatrixOBJ distances =  ExcludeByRadius.convertJsonToDMObject(ExcludeByRadius.distanceMatrixJsonURL(userLocation, locations));
```
Comment (zrideaux) [inline src/main/java/com/google/sps/sort/recommended/RecommendedSort.java:85]: Maybe wrap line 85 after convertJsonToDMObject opens 
```java
    DistanceMatrixOBJ distances =  ExcludeByRadius.convertJsonToDMObject(
        ExcludeByRadius.distanceMatrixJsonURL(userLocation, locations));

```

File: src/main/java/com/google/sps/utility/ExcludeByRadius.java [hunk 3]
```diff
@@ -88,9 +88,13 @@ public static String distanceMatrixJsonURL(String userLocation, String[] listing
     userLocation = userLocation.replace(" ", "+");
     String completeURL = baseURL+"origins="+userLocation+"&destinations=";
     for (int i = 0; i < listingLocations.length; i++) {
+      listingLocations[i] = listingLocations[i].replace(" ", "");
       completeURL += listingLocations[i] + "|";
     }
-    completeURL += "&departure_time=now&key=" + API_KEY; 
+    completeURL += "&departure_time=now&key=" + API_KEY;
+
+    System.out.println(completeURL);
```
Comment (zrideaux) [inline src/main/java/com/google/sps/utility/ExcludeByRadius.java:96]: Reminder to remove log whenever you finish testing!

File: src/main/java/com/google/sps/sort/recommended/RecommendedSort.java [hunk 4]
```diff
@@ -48,17 +53,64 @@
...
+    String[] locations = new String[listings.size()];
+    for (int i = 0; i < listings.size(); i++) {
+      locations[i] = listings.get(i).getLocation(); 
+    }
+    
+    DistanceMatrixOBJ distances =  ExcludeByRadius.convertJsonToDMObject(
+        ExcludeByRadius.distanceMatrixJsonURL(userLocation, locations));
+    
+    int[] distanceValues = distances.getIntegerDistanceValues();
+
```
Comment (kev1n80) [inline src/main/java/com/google/sps/sort/recommended/RecommendedSort.java:89]: There is an extra space here.

**Pull request merged**